### PR TITLE
Create build plugins report script

### PR DESCRIPTION
**Pull request by camilaibs**
## Hey, I just made a Pull Request!

Draft a script for listing plugins' latest update date:
```sh
yarn build:plugins-report
```
image

#### :heavy_check_mark: Checklist

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [x] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: scripts/build-plugins-report.js [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+
+async function main() {
+  const pluginsDirectory = path.resolve(rootDirectory, 'plugins');
+
+  const directoryNames = fs
+    .readdirSync(pluginsDirectory, { withFileTypes: true })
+    .filter(dirent => dirent.isDirectory())
+    .map(dirent => dirent.name);
+
+  const tableRows = [['Plugin', 'Author', 'Message', 'Date']];
```
Comment (camilaibs) [inline scripts/build-plugins-report.js:78]: Would you like me to sort the results by date?

File: scripts/build-plugins-report.js [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+      path.resolve(directoryPath, 'package.json'),
+    );
+
+    const plugin = PLUGIN_ROLES.includes(packageJson?.backstage?.role ?? '');
+
+    if (!plugin) continue;
+
+    let data;
+    let skip = 0;
+    const maxCount = 10;
```
Comment (camilaibs) [inline scripts/build-plugins-report.js:95]: Would you like me to make this configurable via flag?

File: scripts/build-plugins-report.js [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+
+        // exclude commits authored by a bot
+        if (
+          line.startsWith('renovate[bot]') ||
+          line.startsWith('github-actions[bot]')
+        ) {
+          return false;
+        }
+
+        return true;
```
Comment (camilaibs) [inline scripts/build-plugins-report.js:123]: Maybe exclude maintainers' commits as well? I could also expose a regex flag if that works best 🙂

File: scripts/build-plugins-report.js [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+  for (const directoryName of directoryNames) {
+    console.log(`🔎 Reading data for ${directoryName}`);
+
+    const directoryPath = path.resolve(pluginsDirectory, directoryName);
+
+    const packageJson = await fs.readJson(
+      path.resolve(directoryPath, 'package.json'),
+    );
+
+    const plugin = PLUGIN_ROLES.includes(packageJson?.backstage?.role ?? '');
```
Comment (camilaibs) [inline scripts/build-plugins-report.js:89]: Should we include modules?

Comment (camilaibs) [reply scripts/build-plugins-report.js]: Done 🙌🏻 : https://github.com/backstage/backstage/pull/20487/commits/a8aaef7674d6a75095d302c15b75232f2813bc3f

File: scripts/build-plugins-report.js (hunk 3 above)
Comment (camilaibs) [reply scripts/build-plugins-report.js]: I did the first and third refinements, and the data seems to be more accurate, I'll try out the second one.

File: scripts/build-plugins-report.js (hunk 1 above)
Comment (camilaibs) [reply scripts/build-plugins-report.js]: Done 🙌🏻 : https://github.com/backstage/backstage/pull/20487/commits/a8aaef7674d6a75095d302c15b75232f2813bc3f

File: scripts/build-plugins-report.js (hunk 3 above)
Comment (camilaibs) [reply scripts/build-plugins-report.js]: Done 🙌🏻 : https://github.com/backstage/backstage/pull/20487/commits/352ef85d7f74fdb5ae0292972cd91fc0424f6c10

**Review by Rugvip — APPROVED** [review]
Nice! 👍 

Let's :shipit:

File: scripts/build-plugins-report.js [hunk 5]
```diff
@@ -0,0 +1,183 @@
...
+  });
+
+  if (stderr) {
+    console.error(stderr);
+  }
+
+  return stdout.trim();
+}
+
+function findLatestValidCommit(commits, directoryPath) {
```
Comment (Rugvip) [inline scripts/build-plugins-report.js:40]: I think it'd be useful to filter out shared commits too but let's do that in a followup. I could hack on that later today unless you want to call dibs 😁

Comment (camilaibs) [reply scripts/build-plugins-report.js:40]: Amazing, go for it, and thanks for all your ideas 🙌🏻

**Pull request merged**